include/aspect/material_model/interface.h

```diff
@@ -169,6 +177,16 @@ namespace aspect
        * u^T) - \frac 13 \nabla \cdot \mathbf u \mathbf 1$.
        */
       std::vector<SymmetricTensor<2,dim> > strain_rate;
+
+      /**
+       * Optional reference to the cell that contains these quadrature
+       * points. This allows for evaluating properties at the cell vertices
+       * and interpolating to the quadrature points. Note that not all
```

Comment:
"..., or to query the cell for material ids, neighbors, or other information that is not available solely from the locations."